Examples/Algorithms/Geant4/src/ParticleKillAction.cpp

```diff
@@ -49,4 +49,27 @@ void ActsExamples::ParticleKillAction::UserSteppingAction(const G4Step* step) {
                << isSecondary);
     track->SetTrackStatus(G4TrackStatus::fStopAndKill);
   }
+
+  // store the status of the particle
+  // check if we have a particle assigned to track
+  if (eventStore().trackIdMapping.find(track->GetTrackID()) ==
+      eventStore().trackIdMapping.end()) {
+    return;
+  }
```

Comment:
usually I am also a fan of early returns but in this case I think it would be better to move the block below into the if and invert it as people might want to add code at the bottom

Reply:
Done.